libsplinter/src/network/auth/handlers/builder.rs

```diff
@@ -223,7 +223,9 @@ impl AuthorizationDispatchBuilder {
             let signers_to_use = match &self.local_authorization {
                 Some(ConnectionAuthorizationType::Challenge { public_key }) => {
                     let signer = signers.iter().find(|signer| match signer.public_key() {
-                        Ok(signer_public_key) => signer_public_key.as_slice() == public_key,
+                        Ok(signer_public_key) => {
+                            signer_public_key.as_slice() == public_key.as_slice()
```

Comment:
Since `PublicKey` implements `PartialEq`, can we just do the comparison on the public keys?

Reply:
This is comparing splinter::public_key::PublicKey against a cylinder::PublicKey